packages/components/spec/components/validation/Validation.spec.tsx

```diff
@@ -3,48 +3,55 @@ import * as React from 'react';
 import { DatePicker, TextField, Validation } from '../../../src/components';
 import { Validators } from '../../../src/core/validators/validators';
 
-describe('Validation Component', () => {
+function currentEventLoopEnd() {
+  return new Promise(resolve => setImmediate(resolve));
+}
+
+fdescribe('Validation Component', () => {
   describe('Validation test suite => ', () => {
     afterEach(() => {
       jest.restoreAllMocks();
     });
     it('if a validator is present no error message appears before modified', async () => {
+      /** TESTS SHOULD NOT TARGET INTERNALS */
       const validate = jest.spyOn(Validation.prototype, 'updateState');
       const wrapper = shallow(
         <Validation validator={Validators.Required} errorMessage={'Required'}>
           <TextField />
         </Validation>
       );
       expect(wrapper.length).toEqual(1);
-      expect(wrapper.find('.tk-validation--errors').exists()).toBeFalsy();
+      expect(wrapper.find('.tk-validation--error').exists()).toBeFalsy();
       expect(wrapper.find('.tk-validation__errors').exists()).toBeFalsy();
       const mockEvent = { target: { value: 'This is just for test' } };
       wrapper.find('TextField').simulate('change', mockEvent);
+      await currentEventLoopEnd();
+      await currentEventLoopEnd();
       expect(validate).toHaveBeenCalledWith(mockEvent.target.value);
     });
     it('"onValidationChanged" callback should be called on validation', async () => {
       const zone = {
-        onChange: () => null,
-        onValidationChanged: () => null,
         validator: Validators.Required,
       };
-      const change = jest.spyOn(zone, 'onChange');
-      const validate = jest.spyOn(Validation.prototype, 'updateState');
-      const valChange = jest.spyOn(zone, 'onValidationChanged');
+
+      const change = jest.fn();
+      const valChange = jest.fn();
       const wrapper = shallow(
         <Validation
           validator={zone.validator}
           errorMessage={'Required'}
-          onValidationChanged={zone.onValidationChanged}
+          onValidationChanged={valChange}
         >
-          <TextField onChange={zone.onChange} />
+          <TextField onChange={change} />
         </Validation>
       );
       const mockEvent = { target: { value: 'This is just for test' } };
       wrapper.find('TextField').simulate('change', mockEvent);
       expect(change).toHaveBeenCalledWith(mockEvent);
-      await validate;
-      expect(valChange).toHaveBeenCalledWith(true, null);
+      await currentEventLoopEnd();
```

Comment:
Curiosity: What is the rational of the number of `await currentEventLoopEnd()` in this workaround?

Reply:
It's related to debouncing the update method
I think the more it debounces the more we need to wait for it to settle :/